syntax = "proto3";

package protodb.server;

message Insert {
  string database = 1;
  string collection = 2;
  bytes data = 3;
}

message InsertResponse {
  bool success = 1;
  uint64 _id = 2;

  enum FailureCode {
    INVALID_DB = 0;
    INVALID_COLLECTION = 1;
  }
  FailureCode failure_code = 3;
}
